### PR TITLE
Add automatic changelog to the repo

### PR DESCRIPTION
**Pull request by irux**

### CODE REVIEW

**Comment by sujuka99** [comment]
Could you provide an example how the changelog would look?

File: .github/changelog-config.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "categories": [
+        {"title": "## 🚀 Features", "labels": ["feature", "feat", "enhancement"]},
+        {"title": "## 🐛 Fixes", "labels": ["fix", "bug"]},
+        {"title": "## 🧪 Dependencies", "labels": ["dependency"]}
```
Comment (disrupted) [inline .github/changelog-config.json:5]:
```suggestion
        {"title": "## 🧪 Dependencies", "labels": ["type/update"]}
```

File: .github/changelog-config.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+    "categories": [
+        {"title": "## 🚀 Features", "labels": ["feature", "feat", "enhancement"]},
+        {"title": "## 🐛 Fixes", "labels": ["fix", "bug"]},
```
Comment (disrupted) [inline .github/changelog-config.json:4]:
```suggestion
        {"title": "## 🐛 Fixes", "labels": ["type/bug"]},
```

File: .github/changelog-config.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+    "categories": [
+        {"title": "## 🚀 Features", "labels": ["feature", "feat", "enhancement"]},
```
Comment (disrupted) [inline .github/changelog-config.json:3]:
```suggestion
        {"title": "## 🚀 Features", "labels": ["type/feature", "type/enhancement"]},
```

**Review by disrupted — CHANGES_REQUESTED** [review]
let's use the existing labels

what do you think about another category for type/refactor?

File: .github/changelog-config.json (hunk 1 above)
Comment (irux) [reply .github/changelog-config.json:5]: > let's use the existing labels
> 
> what do you think about another category for type/refactor?



> let's use the existing labels
> 
> what do you think about another category for type/refactor?

Any reason to prefix everything with type/ ? doesn't make it redundant ? just the suffix make it clear it is a category or not? what do you think ?

Comment (disrupted) [reply .github/changelog-config.json:5]: that's just what we have at the moment, @raminqaf created the category system

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -35,6 +35,26 @@ jobs:
         uses: actions/checkout@v3
         with:
           persist-credentials: false
+          fetch-depth: 0
+
+      - name: Create changelog
+        id: build_changelog
+        uses: bakdata/ci-templates/actions/changelog-generate@fix/bug-changelog
```
Comment (disrupted) [inline .github/workflows/release.yaml:42]: TODO

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -35,6 +35,26 @@ jobs:
...
+        with:
+          github-token: ${{ secrets.GH_TOKEN }}
+          config: "./.github/changelog-config.json"
+          new-tag: ${{ needs.release-to-pypi.outputs.release-tag }}
+          fetchReviewers: "true"
```
Comment (MichaelKora) [inline .github/workflows/release.yaml:47]:
```suggestion
          fetch-reviewers: "true"
```

File: .github/workflows/release.yaml [hunk 6]
```diff
@@ -35,6 +35,26 @@ jobs:
...
+          fetchReleaseInformation: "true"
```
Comment (MichaelKora) [inline .github/workflows/release.yaml:48]:
```suggestion
          fetch-release-information: "true"
```

File: .github/workflows/testing-changelog.yaml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+      - name: Create changelog
+        id: build_changelog
+        uses: bakdata/ci-templates/actions/changelog-generate@fix/bug-changelog
+        with:
+          github-token: ${{ secrets.GH_TOKEN }}
+          config: "./.github/changelog-config.json"
+          new-tag: 1.1.5
+          fetchReviewers: "true"
+          fetchReleaseInformation: "true"
```
Comment (MichaelKora) [inline .github/workflows/testing-changelog.yaml:30]:
```suggestion
          fetch-release-information: "true"
```

File: .github/workflows/testing-changelog.yaml [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+          fetch-depth: 0
+
+      - name: Create changelog
+        id: build_changelog
+        uses: bakdata/ci-templates/actions/changelog-generate@fix/bug-changelog
+        with:
+          github-token: ${{ secrets.GH_TOKEN }}
+          config: "./.github/changelog-config.json"
+          new-tag: 1.1.5
+          fetchReviewers: "true"
```
Comment (MichaelKora) [inline .github/workflows/testing-changelog.yaml:29]:
```suggestion
          fetch-reviewers: "true"
```

**Review by MichaelKora — CHANGES_REQUESTED** [review]

**Pull request closed without merging**